src/afrolabs/components/http.clj

```diff
@@ -128,5 +148,39 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+;; The resource file identified with `version-info-resource` has to have this format:
+;; {:git-ref  "refs/heads/pwab/BSA-103-add-git-commit-branch-info-into-deployed-artifact-so-you-can-see-what-version-is-running"
+;;  :git-sha  "f7ba5fa0d8bfd1c8a078cae583f8b4cda39b59c3"}
+
 
+(s/def ::version-info-resource string?)
+
+(s/def ::git-version-middleware-cfg
+  (s/keys :req-un [::version-info-resource]
+          :opt-un []))
 
+(defn create-git-version-middleware
+  [{:as   _cfg
+    :keys [version-info-resource]}]
+  (let [{:keys [git-ref git-sha]} (-version/read-version-info version-info-resource)]
+    (reify
+      IRingMiddlewareProvider
+      (get-middleware [_]
+        (if-not (and git-ref git-sha)
+          (do (log/warn (str "Failed to load version info from resource '"
+                             version-info-resource
+                             "'. Unable to create version middleware."))
+              [])
```

Comment:
- load git version from provided resource file, use the recently-added `-version/read-version-info` helper for this.
- when no git data can be loaded, provide a warning